apps/permissions.go

```diff
@@ -46,8 +51,28 @@ func (p Permission) Markdown() md.MD {
 		m = "Use Mattermost REST API as connected users"
 	case PermissionActAsBot:
 		m = "Use Mattermost REST API as the app's bot user"
+	case PermissionRemoteOAuth2:
+		m = "Use remote (3rd party) OAuth2 and store secrets"
 	default:
 		m = "unknown permission: " + string(p)
 	}
 	return md.MD(m)
 }
+
+func (p Permissions) IsValid() error {
+	for _, pp := range []Permissions{
+		{PermissionRemoteOAuth2, PermissionActAsBot, PermissionActAsUser},
+		{PermissionUserJoinedChannelNotification, PermissionActAsBot},
```

Comment:
What are these groups?

Reply:
(A, B, C, ...) - A requires (depends on) B, C, ...

Reply:
Makes sense. Please add a comment about that once your work on the docs.